test_tf2/CMakeLists.txt

```diff
@@ -36,92 +35,83 @@ ament_find_gtest()
 
 ament_add_gtest(buffer_core_test test/buffer_core_test.cpp)
 if(TARGET buffer_core_test)
-  ament_target_dependencies(buffer_core_test
-    builtin_interfaces
-    geometry_msgs
-    rclcpp
-    tf2
-    tf2_geometry_msgs
-    tf2_ros
-  )
+  target_link_libraries(buffer_core_test
+    ${builtin_interfaces_TARGETS}
+    ${geometry_msgs_TARGETS}
+    rclcpp::rclcpp
+    tf2::tf2
+    ${tf2_geometry_msgs_TARGETS}
+    tf2_ros::tf2_ros)
 endif()
 
 ament_add_gtest(test_message_filter test/test_message_filter.cpp)
 if(TARGET test_message_filter)
-  ament_target_dependencies(test_message_filter
-    builtin_interfaces
-    geometry_msgs
-    rclcpp
-    tf2
-    tf2_ros
-  )
+  target_link_libraries(test_message_filter
+    ${builtin_interfaces_TARGETS}
+    ${geometry_msgs_TARGETS}
+    rclcpp::rclcpp
+    tf2::tf2
+    tf2_ros::tf2_ros)
 endif()
 
 ament_add_gtest(test_convert test/test_convert.cpp)
 if(TARGET test_convert)
-  ament_target_dependencies(test_convert
-    Eigen3
-    eigen3_cmake_module
-    tf2
-    tf2_bullet
-    tf2_eigen
-    tf2_geometry_msgs
-    tf2_kdl
-  )
+  target_link_libraries(test_convert
+    ${geometry_msgs_TARGETS}
+    tf2::tf2
+    tf2_kdl::tf2_kdl
+    tf2_bullet::tf2_bullet
+    tf2_eigen::tf2_eigen
+    ${tf2_geometry_msgs_TARGETS}
+    Eigen3::Eigen)
 endif()
 
 ament_add_gtest(test_utils test/test_utils.cpp)
 if(TARGET test_utils)
-  ament_target_dependencies(test_utils
-    geometry_msgs
-    tf2
-    tf2_geometry_msgs
-    tf2_kdl
-  )
+  target_link_libraries(test_utils
+    ${geometry_msgs_TARGETS}
+    ${tf2_geometry_msgs_TARGETS}
+    tf2::tf2
+    tf2_kdl::tf2_kdl)
 endif()
 
 add_executable(test_buffer_server test/test_buffer_server.cpp)
 if(TARGET test_buffer_server)
-  ament_target_dependencies(test_buffer_server
-    rclcpp
-    tf2_bullet
-    tf2_ros
-  )
+  target_link_libraries(test_buffer_server
+    tf2_ros::tf2_ros
+    rclcpp::rclcpp)
 endif()
 
 add_executable(test_buffer_client test/test_buffer_client.cpp)
 if(TARGET test_buffer_client)
-  ament_target_dependencies(test_buffer_client
-    rclcpp
-    tf2_bullet
-    tf2_geometry_msgs
-    tf2_kdl
-    tf2_ros
-  )
-  target_link_libraries(test_buffer_client ${GTEST_LIBRARIES})
+  target_link_libraries(test_buffer_client
+    ${GTEST_LIBRARIES}
```

Comment:
Just a question: is there no modern CMake target for GTest?

Reply:
Their might be. I didn't look :no_mouth:. If so it's probably in the `GTEST_LIBRARIES` variable.

Reply:
I suspect the `${GTEST_LIBRARIES}` variable always contains the modern CMake target `gtest`, but I haven't dug through `ament_cmake_gtest` enough to be confident in that. I'll leave this as is for now.